services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/dataset_relevant_text.rb

```
@@ -37,6 +38,14 @@ def serializable_hash(options = nil)
           ))
         end
 
+        def duplicate_for_dataset_parent
+          return if dataset.parent_id.nil?
+
+          duplicate_dataset_relevant_text = DatasetRelevantText.new(relevant_text_id:, visible:)
```

You can collapse these three lines into one; also, I would use `create!` otherwise this will fail silently:

`DatasetRelevantText.create!(relevant_text_id:, visible:, dataset_id: dataset.parent_id)`